docs/core/deploying/runtime-patch-selection.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Self-contained deployment runtime roll forward
+description: Learn about dotnet publish changes for self-contained deployments.
+author: jralexander
+ms.author: johalex
+ms.date: 05/30/2018
+---
+# Self-contained deployment runtime roll forward
+
+.NET Core [self-contained application deployments](index.md) include both the .NET Core libraries and the .NET Core runtime. Starting in .NET Core SDK 2.1.300 (.NET Core 2.1), a self-contained application deployment [now publishes the highest patch runtime on your machine](https://github.com/dotnet/designs/pull/36). By default,[`dotnet publish`](../tools/dotnet-publish.md) for a self-contained deployment selects the latest version installed as part of the SDK on the publishing machine. This enables your deployed application to run with security fixes (and other fixes) available during `publish`. The application must be re-published to obtain a new patch. Self-contained applications are created by specifying `--self-contained` on `dotnet publish` command or by specifying the RuntimeIdentifier in the project file or on the command line.
+
+[`restore`](../tools/dotnet-restore.md), [`build`](../tools/dotnet-build.md) and [`publish`](../tools/dotnet-publish.md) are `dotnet` commands that can run separately. The runtime choice is part of the `restore` operation, not `publish` or `build`. If you call `publish`, the latest patch version will be chosen. If you call `publish` with the `--no-restore` argument, then you may not get the desired patch version because a prior `restore` may not have been executed with the new self-contained application publishing policy.
```

Comment:
> you may not get the desired patch version

If you would get the wrong patch version in this case, you will actually get a build error.  So I'd add the following sentence: 

> In this case, a build error will be generated with text similar to the following:
>> The project was restored using Microsoft.NETCore.App version 2.0.0, but with current settings, version 2.0.6 would be used instead. To resolve this issue, make sure the same settings are used for restore and for subsequent operations such as build or publish. Typically this issue can occur if the RuntimeIdentifier property is set during build or publish but not during restore.

Once this doc is published I think it would be a good idea to add an aka.ms link to this doc page in the error message.

Reply:
Agreed